test/index.jl

```diff
@@ -13,11 +14,11 @@ inds = Any[1,
            [big(1)],
            big(1):big(1),
            [:A],
-           Union{Bool, Missing}[true, false],
```

Comment:
throws deprecation warning for `Union{T,Missing}` vectors since indexing by `missing` is deprecated.  note that this is now tested in "deprecated.jl".

Reply:
On my machine it does not throw deprecation warning. The vector does not contain any missing and `getindex` for vector of `Union{Bool, Missing}` tests for it. And this specific case is not tested in "deprecated.jl".

Reply:
Yes this should not have been removed, not sure why I did, will restore.  I'm actually pretty dubious about whether it makes sense to accept vectors of this type, but that's a separate issue so I'll file one.

Reply:
Agreed 😄. When I was making a clean up of `getindex` I also wondered why we accept such unions, but we did so I left it (probably this is due to the fact that in some cases the indexing vector might be dynamically generated and some methods "by default" can create such unions without checking if they are needed).

Reply:
Yes, I think it makes sense to ignore the type of the array and look only at the actual type of the elements. Else it can be cumbersome given that Julia is a dynamic language. That's how indexing arrays works.

Reply:
The only exception to keep in mind is the fact that `Bool <: Integer` and we want to handle `Bool` in a different way than `Integer`. It is now covered correctly by `getindex` (i.e. array consisting only of bool values uses boolean indexing only if its type is `AbstractVector{Bool}` or `AbstractVector{Union{Missing,Bool}}` and otherwise it is an error) but if someone changes this code it has to be kept in mind.

Reply:
The only way to ensure this doesn't regress is to add tests...

Reply:
Agreed. @ExpandingMan could you please make sure that we have the following tests?
Otherwise I can add it later, but maybe it is better to fix it in one shot.

Tests that should pass after deprecation:
```
[true, false]
Union{Missing,Bool}[true, false]
[false:false]
false:true
```
(the last one is a corner case but is OK in this particular case)

Tests that should fail:
```
[true]
[true, false, true]
[true, 1]
Any[true, true]
Integer[true, false]
[missing, true]
Union{Missing, Bool}[missing, missing]
```

Reply:
`[true, 1]` should be `Any[true, 1]` or it's equivalent to `[1, 1]`.